### ISSUE TITLE
"kill" command doesn't work with shim/proxy

### ISSUE THREAD

**Issue by devimc** [01org/cc-oci-runtime#455]
we have different ways to fix this:
* delete command should stop to cc-shim
* cc-proxy should notify to cc-shim when the Vm ends
* cc-shim should monitor to the VM

**Comment by dlespiau** [comment]
So, I don't think delete should stop cc-shim. delete should only be called when cc-shim has already exited

**Comment by devimc** [comment]
I applied this patch #460 and then ```create -> start -> kill -> delete``` a container
cc-shim became crazy printing a lot of messages
```
cc-shim[14691]: read_IO_message:377:EOF received on proxy I/O fd
cc-shim[14691]: read_IO_message:377:EOF received on proxy I/O fd
.....
```

**Comment by chavafg** [comment]
Not sure if this is related to this issue.

I got a cc-shim process that was not stopped after the container was shutdown.
I killed it with `killall cc-shim` and `cc-proxy` was also killed. is this a correct behavior? 

stderror from cc-proxy when it was killed: 
```
+ ./cc-proxy
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x58 pc=0x40459a]

goroutine 635 [running]:
panic(0x526b80, 0xc42000c130)
        /usr/lib/golang/src/runtime/panic.go:500 +0x1a1
main.(*vm).writeMessage(0xc4203d0380, 0xc42003cd80, 0xc400000012, 0x0)
        /root/go/src/github.com/01org/cc-oci-runtime/proxy/vm.go:150 +0x15a
main.(*vm).SendMessage(0xc4203d0380, 0xc4200742a0, 0xd, 0xc4200700c0, 0x5d, 0x60, 0x0, 0x0)
        /root/go/src/github.com/01org/cc-oci-runtime/proxy/vm.go:350 +0x136
main.hyperHandler(0xc4200b8090, 0x82, 0x90, 0x513ae0, 0xc4203a9ce0, 0xc4203a9fa0)
        /root/go/src/github.com/01org/cc-oci-runtime/proxy/proxy.go:198 +0x154
main.(*protocol).handleRequest(0xc420082020, 0xc42003cf20, 0xc4203baab0, 0xc4203a9fa0, 0x0)
        /root/go/src/github.com/01org/cc-oci-runtime/proxy/protocol.go:95 +0xca
main.(*protocol).Serve(0xc420082020, 0x5e0940, 0xc42002b6f8, 0x513ae0, 0xc4203a9ce0)
        /root/go/src/github.com/01org/cc-oci-runtime/proxy/protocol.go:130 +0x1ad
main.(*proxy).serveNewClient(0xc420084100, 0xc420082020, 0x5e0940, 0xc42002b6f8)
        /root/go/src/github.com/01org/cc-oci-runtime/proxy/proxy.go:250 +0xb5
created by main.(*proxy).serve
        /root/go/src/github.com/01org/cc-oci-runtime/proxy/proxy.go:270 +0x3eb
root@two-dot-one ~ # 
```

**Comment by dlespiau** [comment]
proxy encountering an error and exiting can never be correct behaviour :/ mind filling another bug for that one?

**Comment by chavafg** [comment]
sure, opening another issue.

**Comment by dlespiau** [comment]
mind telling me what's at:

```
        /root/go/src/github.com/01org/cc-oci-runtime/proxy/vm.go:150 +0x15a
```

That's not the latest code you are running (also, working as root!!)

**Comment by sameo** [comment]
This should be fixed with PR #495
